### PR TITLE
Change test_grdcontour.py to use static_earth_relief

### PR DESCRIPTION
**Pull request by willschlitzer**
This modifies `test_grdcontour.py` to use `load_static_earth_relief`.

Addresses #1684 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by meghanrjones** [comment]
> @GenericMappingTools/pygmt-maintainers Does `test_grdcontour.py` need to keep `test_grdcontour_file()`? I feel like it's not testing anything new, as the import functions are already tested in other functions, and plotting is tested by the other functions.

I agree with you that it's not needed.

**Review by meghanrjones — COMMENTED** [review]
The region in `test_grdcontour_interval_file_full_opts` also needs adjusting.

File: pygmt/tests/test_grdcontour.py [hunk 1]
```diff
@@ -88,14 +90,14 @@ def test_grdcontour_interval_file_full_opts(grid):
...
-        "region": [-161.5, -154, 18.5, 23],
+        "region": [-53, -49, -20, -17],
         "interval": TEST_CONTOUR_FILE,
         "grid": grid,
         "resample": "100",
         "projection": "M10c",
         "cut": 10,
     }
-    fig.grdcontour(**comargs, limit=(-25000, -1), pen=["a1p,blue", "c0.5p,blue"])
+    fig.grdcontour(**comargs, limit=(0, 400), pen=["a1p,blue", "c0.5p,blue"])
```
Comment (meghanrjones) [inline pygmt/tests/test_grdcontour.py:100]: Do you think there is a bug here? From how I interpret the docs, `limit=(0, 400)` means that contours above 400 should not be drawn but the image looks like there are contours up to at least 800.

Comment (willschlitzer) [reply pygmt/tests/test_grdcontour.py:100]: No, it's from user error on my part; the next command just had a lower limit of 0, so everything above it was plotted. I rewritten the test.

File: pygmt/tests/test_grdcontour.py [hunk 2]
```diff
@@ -39,10 +41,10 @@ def test_grdcontour_labels(grid):
     """
     fig = Figure()
     fig.grdcontour(
-        grid,
-        interval="1000",
-        annotation="5000",
-        projection="W0/15c",
+        grid=grid,
+        interval="10",
+        annotation="50",
```
[the remark this answers is not in the document]
Comment (willschlitzer) [reply pygmt/tests/test_grdcontour.py]: updated in 3fcbc45

**Review by seisman — APPROVED** [review]

**Pull request merged**